gpytorch/lazy/kronecker_product_lazy_tensor.py

```diff
@@ -112,3 +117,45 @@ def _size(self):
 
     def _transpose_nonbatch(self):
         return self.__class__(*(lazy_tensor._transpose_nonbatch() for lazy_tensor in self.lazy_tensors), **self._kwargs)
+
+    @cached
+    def inverse(self):
+        # here we use that (A \kron B)^-1 = A^-1 \kron B^-1
+        inverses = [lt.inverse() for lt in self.lazy_tensors]
+        return self.__class__(*inverses)
+
+    # TODO: Investigate under what conditions computing individual individual inverses makes sense
+
+
+class KroneckerProductTriangularLazyTensor(KroneckerProductLazyTensor):
+    def __init__(self, *lazy_tensors, upper=False):
+        from .triangular_lazy_tensor import TriangularLazyTensor
+
+        if not all(isinstance(lt, TriangularLazyTensor) for lt in lazy_tensors):
+            raise RuntimeError("Components of KroneckerProductTriangularLazyTensor must be TriangularLazyTensor.")
+        super().__init__(*lazy_tensors)
+        self.upper = upper
+
+    def _cholesky_solve(self, rhs, upper=False):
+        if upper:
+            # res = (U.T @ U)^-1 @ v = U^-1 @ U^-T @ v
+            w = self._transpose_nonbatch().inv_matmul(rhs)
+            res = self.inv_matmul(w)
+        else:
+            # res = (L @ L.T)^-1 @ v = L^-T @ L^-1 @ v
+            w = self.inv_matmul(rhs)
+            res = self._transpose_nonbatch().inv_matmul(w)
+        return res
+
+    @cached
+    def inverse(self):
+        # here we use that (A \kron B)^-1 = A^-1 \kron B^-1
+        inverses = [lt.inverse() for lt in self.lazy_tensors]
+        return self.__class__(*inverses, upper=self.upper)
+
+    def inv_matmul(self, right_tensor, left_tensor=None):
+        # Since the individual tensors are triangular, their inverses can be computed cheaply
+        res = self.inverse() @ right_tensor
```

Comment:
Could you do something without making the explicit inverse here? From https://github.com/cornellius-gp/gpytorch/issues/725: 
```
        tensor_shapes = [q.shape[0] for q in self.lazy_tensors]
        multipliers = torch.Tensor([torch.prod(torch.Tensor([ts for i, ts in enumerate(tensor_shapes) if i!=j]))
                                for j in range(len(tensor_shapes))])

        y = inv_quad_rhs.clone()
        for i, q in enumerate(self.lazy_tensors):
            n = q.shape[0]
            y = q.inv_matmul(y.contiguous().view(n, -1))
            y = y.view(n, nrows//n, -1).permute(1, 0, 2)
```

Reply:
Yeah this seems analogous to what we have for `KPLT._matmul`.

Reply:
Added that in, got slightly more complicated to allow arbitrary batch shapes, but it works. I implemented it generically for `KroneckerProductLazyTensor`, but made sure it's the default only for `KroneckerProductTriangularLazyTensor` for now (in which case it's just a loop over batched triangular solves, so really cheap and should probably always be done).